src/snapred/backend/service/CalibrationService.py

```diff
@@ -196,29 +197,82 @@ def fitPeaks(self, request: FitMultiplePeaksRequest):
 
     @FromString
     def save(self, request: CalibrationExportRequest):
+        """
+        If no version is attached to the request, this will save at next version number
+        """
         entry = request.calibrationIndexEntry
-        version = entry.version
-        calibrationRecord = request.calibrationRecord
-        if version is not None:
-            calibrationRecord.version = version
-        calibrationRecord = self.dataExportService.exportCalibrationRecord(calibrationRecord)
-        calibrationRecord = self.dataExportService.exportCalibrationWorkspaces(calibrationRecord)
-        entry.version = calibrationRecord.version
-        self.saveCalibrationToIndex(entry)
+        record = request.calibrationRecord
+        version = self.dataFactoryService.getThisOrNextCalibrationVersion(
+            record.runNumber,
+            record.useLiteMode,
+            request.version,
+        )
```

Comment:
Use wither the version from the request, or use the Indexor to find the next version.  Needed to rename the workspaces.

Reply:
I think this should explicitly fail if the given version is not writable.  Ambiguity should be bubbled up if it exists.